eyeball-im/src/vector.rs

```diff
@@ -83,8 +83,10 @@ impl<T: Clone + Send + Sync + 'static> ObservableVector<T> {
         #[cfg(feature = "tracing")]
         tracing::debug!(target: "eyeball_im::vector::update", "clear");
 
-        self.values.clear();
-        self.broadcast_diff(VectorDiff::Clear);
+        if !self.values.is_empty() {
+            self.values.clear();
+            self.broadcast_diff(VectorDiff::Clear);
+        }
```

Comment:
I like this, but I think if tracing is enabled, the tracing event should reflect the fact that broadcasting was skipped. Maybe something like this?

```rust
let already_empty = self.values.is_empty();

#[cfg(feature = "tracing")]
tracing::debug!(
    target: "eyeball_im::vector::update",
    nop = already_empty.then_some(true),
    "clear"
);

if !already_empty {
    self.values.clear();
    self.broadcast_diff(VectorDiff::Clear);
}
```

(by using `then_some`, the `nop` field would only get included in the output when `already_empty` is `true`)

Reply:
Done. I've also added a test.